### DISCUSSION TITLE
Relais output on spcific states?

### DISCUSSION THREAD

**Discussion by StefanR71** [raomin/ESPAltherma#389] [Ideas]
Hi,

i'm using this solution now for some time and it woks very well. 

We have here a different heat control unit and it will be great when i can control that with a relay output of the ESP32.

I need to relay outputs to get this states:

"Defrost Operation" ON/OFF
"Operation Mode" When set to cooling reais output -> ON
"Space heating Operation ON/OFF" ON/OFF

Is this easy to program? Or can someone help me to get the right direction?

The reason is, i have state inputs on the other control unit and it will be perfect when i can control that direct. At the moment i query them and set them via script. Perfect will be without this script...

Kind regards
Stefan

**Comment by raomin** [comment]
You can easily do with automations this if your state relays are also in Home Assistant.

**Comment by StefanR71** [comment]
My goal is to make most of the communication directly in hardware without a additional server between the components. It should work even when my server is not working...

**Comment by StefanR71** [comment]
Okay, i tested something...

I changed the code and now i'm able to switch 4 relays on and off via mqtt, so the relay board is working.

Now i must understand how you query the values, then i need to compare the values and if the value has the required content then a relay should change his state.... This i must place somewhere in the loop.

So a litte bit more testing and learning... :-)

**Comment by StefanR71** [comment]
So, i'm nearly finished. I used the ESPAltherma as base to create a new program to fit my needs.

First i got the Daikin HomeHub to get Modbus RTU, then i made this construction:

<img width="534" alt="image" src="https://github.com/raomin/ESPAltherma/assets/90922794/163c4a7a-ddd2-4203-b9a2-c130da367174">

I query every 5 sec the Modbus for the important values and send them via MQTT. It also watches some topics on the MQTT server to switch DHT on/off,  heting/cooling on/off, switching between heating/colling/auto or set the target water temperature and send the commands via Modbus to the heatpump.

And here i need to control a different control unit for the 2 heating circuits, for that i need the relay output to connect them to the sensor input of that unit. One is to switch to cooling mode and one is to switch heating on/off or better the pumps in the circuit.

And the pumps i want to switch off when the heatpump is going to defrost and is in DHT mode...

For me it is the perfect addon to the ESPAltherma to "control" the heatpump.

Perhaps it is an idea to include the Modbus interface to the ASPAltherma project to control the heatpump...

**Comment by StefanR71** [comment]
Okay i'm finished, i have changed a few things, i used now TinyMQTT to have an internal MQTT broker. I decided to do this, to be independed from other servers or external services. Now only when there is no WLAN the unit will not work. It is bad, when i want to control things and it can't work only because the linux server with the MQTT broker isn't available.

And i put all in a case:
![image](https://github.com/raomin/ESPAltherma/assets/90922794/05ff273a-6ec2-430a-87c9-3024a26a4757)

I'm very happy with that. :-)
